### DISCUSSION TITLE
Hiding the tab bar on Firefox 117.0.1

### DISCUSSION THREAD

**Discussion by WhyNotHugo** [piroor/treestyletab#3390] [Q&A] [answered]
Advise from the wiki doesn't seem to work: https://github.com/piroor/treestyletab/wiki/Code-snippets-for-custom-style-rules#hide-horizontal-tabs-at-the-top-of-the-window-1349-1672-2147

It hides all tabs, but keeps around the empty space from the tab bar:

![image](https://github.com/piroor/treestyletab/assets/730811/d5fa1e04-e473-454c-a745-cb93ab512e00)

```console
> cat profile/chrome/userChrome.css 
#main-window[tabsintitlebar="true"]:not([extradragspace="true"]) #TabsToolbar > .toolbar-items {
  opacity: 0;
  pointer-events: none;
}
#main-window:not([tabsintitlebar="true"]) #TabsToolbar {
    visibility: collapse !important;
}
```

**Comment by WhyNotHugo** [comment, marked as answer]
The following seems to work

```css
#TabsToolbar {
  visibility: collapse;
}
```

Source: https://superuser.com/a/1424494

**Comment by irvinm** [comment]
There is information on the wiki as well (including the answer you already found):  https://github.com/piroor/treestyletab/wiki/Code-snippets-for-custom-style-rules#hide-horizontal-tabs-at-the-top-of-the-window-1349-1672-2147

**Reply by WhyNotHugo** [reply]
The link that you've just posted in the exact same link that I included in my original question. The information in there didn't work for me on 117.0.1. It hides all tabs, but keeps around the empty space from the tab bar, as seen in the screenshot above.

**Reply by WhyNotHugo** [reply]
You probably want to incorporate the details from my answer above into the wiki -- although I'm unsure at which version of Firefox things changed.

**Reply by irvinm** [reply]
Yes, you are right ... that section looks to be out of date.  Feel free to update that section in the wiki.  @piroor usually adds major pieces to the wiki but asks for everyone help out with updates to it.  I just checked my solution and it is similar to yours but is more complex to support hiding and showing of the tab bar dynamically.  However, I do see that the my solution also involves "#TabsToolbar" as you point out in the solution.

**Comment by irvinm** [comment]
So I went back and looked at the older solution (as I might have used it in the past) and loaded up a new profile to test some things out.

I used the original solution and see the same thing you do with the extra bar with the window controls.  Technically I guess the tabs are removed, but not an elegant solution.

![image](https://github.com/piroor/treestyletab/assets/979729/480d76fe-7205-4d1f-8afd-37afe89efea9)

I had an older alternate solution that basically used the same initial solution and then SHIFTED down the nav bar controls down into the tab bar and shifted the right-most items to the left.  So if you wanted to KEEP the window controls you can try something like this. (Obviously you would need to customize for your instance in terms of how much to shift things)

![image](https://github.com/piroor/treestyletab/assets/979729/9c9120d2-812f-41af-9d6f-6c314b12fb99)

I assume you see something like this with the proposed solution?

![image](https://github.com/piroor/treestyletab/assets/979729/0ddadeeb-0f89-4cf5-b2ca-400f216a7227)
